oozie-to-airflow/mappers/mapreduce_mapper.py

```diff
@@ -52,15 +52,17 @@ def __init__(
         self.params_dict = {}
         self.file_extractor = FileExtractor(oozie_node=oozie_node, params=params)
         self.archive_extractor = ArchiveExtractor(oozie_node=oozie_node, params=params)
-        self._parse_oozie_node()
+        self.name_node = None
+        self.hdfs_files = None
+        self.hdfs_archives = None
 
-    def _parse_oozie_node(self):
+    def on_parse_node(self):
         name_node_text = self.oozie_node.find("name-node").text
         self.name_node = el_utils.replace_el_with_var(name_node_text, params=self.params, quote=False)
         self._parse_config()
         self._parse_params()
-        self.files, self.hdfs_files = self.file_extractor.parse_node()
-        self.archives, self.hdfs_archives = self.archive_extractor.parse_node()
+        _, self.hdfs_files = self.file_extractor.parse_node()
```

Comment:
Just a note - we should in a separate PR to move extractor use up to the mixin - there is no reason it should be used inside each mapper separately. But not now.

Reply:
I have doubts about whether it should be a class. As a function it will be better to fulfill its role, it will be clearer and simpler.